sympy/series/fourier.py

```diff
@@ -0,0 +1,391 @@
+                % func)
+
+    x, start, stop = None, None, None
+    if limits is None:
+        x, start, stop = _find_x(func), -pi, pi
+    if is_sequence(limits, Tuple):
+        if len(limits) == 3:
+            x, start, stop = limits
+        elif len(limits) == 2:
+            x = _find_x(func)
+            start, stop = limits
+
+    if not isinstance(x, Symbol) or start is None or stop is None:
+        raise ValueError('Invalid limits given: %s' % str(limits))
+
+    unbounded = [S.NegativeInfinity, S.Infinity]
+    if start in unbounded or stop in unbounded:
+            raise ValueError("Both the start and end value"
+                             " should be bounded")
+
+    return sympify((x, start, stop))
+
+
+class FourierSeries(SeriesBase):
+    r"""Represents fourier sine/cosine series
+
+    Examples
+    ========
+
+    >>> from sympy import FourierSeries, pi, cos
+    >>> from sympy.abc import x
+
+    >>> s = FourierSeries(x**2, (x, -pi, pi))
+    >>> s.as_series(n=3)
+    -4*cos(x) + cos(2*x) + pi**2/3
+
+    Shifting
+
+    >>> s.shift(1)
+    FourierSeries(x**2 + 1, (x, -pi, pi))
+    >>> s.shiftx(1)
+    FourierSeries((x + 1)**2, (x, -pi, pi))
+
+    Scaling
+
+    >>> s.scale(2)
+    FourierSeries(2*x**2, (x, -pi, pi))
+    >>> s.scalex(2)
+    FourierSeries(4*x**2, (x, -pi, pi))
+
+    Differentiating
+
+    >>> s.diff(x)
+    FourierSeries(2*x, (x, -pi, pi))
+
+    Notes
+    =====
+
+    Computing a fourier series can be slow
+    due to the integration required in computing
+    an, bn.
+
+    It is faster to compute fourier series of a function
+    by using shifting and scaling on an already
+    computed fourier series rather than computing
+    again.
+
+    eg. If Fourier series of x**2 is known
+    fourier series of x**2 - 1 can be found by shifting by 1
+
+    References
+    =========
+
+    .. [1] mathworld.wolfram.com/FourierSeries.html
+    """
+    def __new__(cls, func, limits=None, **kwargs):
+        func = sympify(func)
+
+        limits = _process_limits(func, limits)
+        x, lower, upper = limits
+
+        if x not in func.free_symbols:
+            return func
+
+        kan = kwargs.pop('an', None)
+        kbn = kwargs.pop('bn', None)
+
+        if kan is None and kbn is None:
+            n = Dummy('n')
+            neg_func = func.subs(x, -x)
+            if func == neg_func:
+                an = fourier_cos_seq(func, limits, n)
+                bn = SeqFormula(0, (1, oo))
+            elif func == -neg_func:
+                an = SeqFormula(0, (1, oo))
+                bn = fourier_sin_seq(func, limits, n)
+            else:
+                an = fourier_cos_seq(func, limits, n)
+                bn = fourier_sin_seq(func, limits, n)
+        else:
+            an, bn = kan, kbn
+
+        a0 = kwargs.pop('a0', None)
+        if a0 is None:
+            a0 = an[0] / 2
+
+
+        obj = Expr.__new__(cls, func, limits)
```

Comment:
You also need to pass `an` and `bn` into the args. SymPy rebuilds objects (e.g. on a `.subs()`) from the args, whatever you pass to the args should be able to re-create the same object.

That said, I'm not sure how `an` and `bn` are supposed to work as kwargs, is it supposed to override the computation? should you be able to pass `an` and `bn` instead of `func`? it seems to not have any effect if I pass them.